app/lib/hooks/useGit.ts

```diff
@@ -101,65 +122,124 @@ export function useGit() {
 const getFs = (
   webcontainer: WebContainer,
   record: MutableRefObject<Record<string, { data: any; encoding?: string }>>,
+  counter: MutableRefObject<{ total: number; completed: number }>,
 ) => ({
   promises: {
```

Comment:
i think we should make these two variable optional 
`record` and `counter`
and use object as argument like 
```typescript
const getFs = (options:{
  webcontainer: WebContainer;
  record: MutableRefObject<Record<string, { data: any; encoding?: string }>>;
  counter: MutableRefObject<{ total: number; completed: number }>;
  }
)=>{...}
```